core/services/relay/evm/chain_reader.go

```diff
@@ -173,14 +173,26 @@ func (cr *chainReader) Unbind(ctx context.Context, bindings []commontypes.BoundC
 func (cr *chainReader) GetLatestValue(ctx context.Context, readName string, confidenceLevel primitives.ConfidenceLevel, params any, returnVal any) error {
 	binding, address, err := cr.bindings.GetReader(readName)
 	if err != nil {
+		cr.lggr.Debug(err.Error())
+
 		return err
 	}
 
-	return binding.GetLatestValue(ctx, common.HexToAddress(address), confidenceLevel, params, returnVal)
+	err = binding.GetLatestValue(ctx, common.HexToAddress(address), confidenceLevel, params, returnVal)
+	if err != nil {
+		cr.lggr.Debug(err.Error())
```

Comment:
I'm not a fan of this pattern since it leads to double reporting of errors. What is the use case for logging the error that we are already returning? How will a node op know that this is not a distinct error from the one returned, if it ends up being logged as well?

Reply:
Alternatively, we could generalize this a bit to be a GRPC "interceptor" that can cover every service, rather than just chain reader. This avoids the double error problem, because all of the interceptor logs come from the same place and the same logger name, so it is clear that this is a middleman and not the source or the handler of the error.

Reply:
I like that solution better than logging directly in the implementation of `ContractReader`.

Reply:
Client side dial options here: https://github.com/smartcontractkit/chainlink-common/blob/1b0938c4678b7a4bae8f8fb93cda2a66cb90bdff/pkg/loop/telem.go#L163
Server side here: https://github.com/smartcontractkit/chainlink-common/blob/1b0938c4678b7a4bae8f8fb93cda2a66cb90bdff/pkg/loop/telem.go#L191

Reply:
I don’t think logging like this is necessary. I was considering whether it would make sense to always log internal errors, while returning a generic internal error message to the caller. For errors caused by the caller, we could return the specific error message directly without logging so that the caller can fix the issue. But if we are always surfacing the whole error then there is no need to double log.